internal/appconfig/config.go

```diff
@@ -132,10 +132,13 @@ type Static struct {
 }
 
 type Mount struct {
-	Source      string   `toml:"source,omitempty" json:"source,omitempty"`
-	Destination string   `toml:"destination,omitempty" json:"destination,omitempty"`
-	InitialSize string   `toml:"initial_size,omitempty" json:"initial_size,omitempty"`
-	Processes   []string `toml:"processes,omitempty" json:"processes,omitempty"`
+	Source                 string   `toml:"source,omitempty" json:"source,omitempty"`
+	Destination            string   `toml:"destination,omitempty" json:"destination,omitempty"`
+	InitialSize            string   `toml:"initial_size,omitempty" json:"initial_size,omitempty"`
+	Processes              []string `toml:"processes,omitempty" json:"processes,omitempty"`
+	ExtendThresholdPercent int      `toml:"extend_threshold_percent,omitempty" json:"extend_threshold_percent,omitempty"`
+	AddSizeGb              int      `toml:"add_size_gb,omitempty" json:"add_size_gb,omitempty"`
+	SizeGbLimit            int      `toml:"size_gb_limit,omitempty" json:"size_gb_limit,omitempty"`
```

Comment:
Can we come up with a naming scheme that reflects all this fields are related to size extension ?

* `extend_size_limit`
* `extend_size_increment`
* `extend_size_threshold`

Another nit is that we recently started supporting more human parseable values for sizes, see how it is done for `initial_size` and machine's `--vm-memory` flags for example.

I can do all these changes if you don't have time for it.

Reply:
Since both limits are GB we might just append in the name `_gb`

How about:

- `auto_extend_add_gb`
- `auto_extend_limit_gb`
- `auto_extend_threshold`

Reply:
nope, the idea is that the field type is a string and accepts units like in `10gb` or `1024mb` and defaults to GB if none is passed. See how it is done for the keys I mentioned.

Reply:
I see. Will need to also validate there's no fractional GB too. You names make sense now

Reply:
`autoextend` as a prefix makes sense tbh:

* `autoextend_size_limit`
* `autoextend_size_increment`
* `autoextend_size_threshold`

Reply:
Just seen this now. How about `auto_extend_*`?

Reply:
done